src/main/java/online/book/store/service/impl/BookServiceImpl.java

```diff
@@ -37,4 +37,17 @@ public BookDto findById(Long id) {
         );
         return bookMapper.toDto(book);
     }
+
+    @Override
+    public BookDto update(CreateBookRequestDto requestDto, Long id) {
+        Book existingBook = bookRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Can't find book with id: " + id));
+        bookMapper.updateBookFromDto(requestDto, existingBook);
+        return bookMapper.toDto(bookRepository.save(existingBook));
```

Comment:
Add `@Transactional` above the class and then you can avoid calling the save() method. Hibernate will perform the update using the dirty-checking mechanism.
```suggestion
        return bookMapper.toDto(existingBook);
```